src/PostgREST/DbRequestBuilder.hs

```diff
@@ -46,13 +46,13 @@ readRequest maxRows allRels allProcs apiRequest  =
       let target = iTarget apiRequest in
       case target of
```

Comment:
This is getting better. I tried to simplify it to the following but it causes two test failures:

```haskell
case target of
  (TargetIdent (QualifiedIdentifier s t) ) ->
    Just (s, t)
  (TargetProc  (QualifiedIdentifier s proc) ) ->
    Just (s, fromMaybe proc $ pdName <$> M.lookup proc allProcs)
  _ -> Nothing
```

Is `pdName` sometimes getting set to the incorrect value? The two test failures talk about `pg_source`, like "No relation found between pg_source and client" and "No relation found between pg_source and articleStars."